app/controllers/uploads_controller.rb

```diff
@@ -20,10 +20,14 @@ def show
   def destroy
     @upload = Upload.find(params[:id])
     @upload.destroy
-    if current_user.admin?
+    if current_user.admin? || current_user.id == @upload.folder.owner.id
       flash[:danger] = "File deleted."
-      redirect_to admin_dashboard_path
-    else  
+      if current_user.admin?
+        redirect_to admin_dashboard_path
+      else
+        redirect_to home_path
+      end 
```

Comment:
I'm starting to really dislike these nested if..else redirects we have throughout our code. Is there some way of refactoring?